spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
> Do you agree that if we don't fail fast, there should be some kind of warning log indicating that initializer was disabled due to particular combination of configuration?

I don't: If you have expressed explicitly that you want your table to be named XYZ, I wouldn't complain if boot will crash down the road because XYZ doesn't exist.

I'd like @wilkinsona to chime in to confirm because I am not 100% sure.